azurerm/resource_arm_notification_hub_authorization_rule.go

```diff
@@ -106,21 +107,24 @@ func resourceArmNotificationHubAuthorizationRuleCreateUpdate(d *schema.ResourceD
 		},
 	}
 
-	if _, err := client.CreateOrUpdateAuthorizationRule(ctx, resourceGroup, namespaceName, notificationHubName, name, parameters); err != nil {
-		return fmt.Errorf("Error creating Authorization Rule %q (Notification Hub %q / Namespace %q / Resource Group %q): %+v", name, notificationHubName, namespaceName, resourceGroup, err)
-	}
+	return resource.Retry(d.Timeout(schema.TimeoutCreate), func() *resource.RetryError {
```

Comment:
since we don't support custom timeouts at this point (it's coming to every resource in 2.0, so the provider behaves consistently) - as such could we hard code this for the moment? e.g.

```suggestion
	return resource.Retry(5 * time.Minute, func() *resource.RetryError {
```

Reply:
Updated with Locks instead, looks cleaner and no retries are needed.

I was thinking about delete but opted not to add more code for the non existing problem as delete does not seem to be affected by this issue as cleanup is always consistent... Anyway updated Delete with the locks as well so should be good.

BTW comment on the azureRMLockByName says deprecated, is there something else to use in future?